File: sentimental-credit/credit.py
from cs50 import get_string
import sys


card_number = get_string("Please enter card number: ")
length = len(card_number)


cursor = length - 1
sum = 0

while cursor >= 1:
    tmp = 2 * int(card_number[cursor - 1])

    if tmp > 9:
        sum += 1 + (tmp % 10)
    else:
        sum += tmp

    cursor -= 2
    tmp = 0


cursor = length - 1

while cursor >= 0:
    sum += int(card_number[cursor])
    cursor -= 2


if sum % 10 != 0:
    print("INVALID")
    sys.exit(1)


n = int(card_number[0:2])

if n == 34 or n == 37 and length == 15:
    print("AMEX")

elif n > 50 and n < 56 and length == 16:
    print("MASTERCARD")

elif n > 39 and n < 50 and (length == 13 or length == 16):
    print("VISA")

else:
    print("INVALID")

File: credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long long creditCardNumber = get_long_long("Please enter credit card number: ");
    int digit1 = 0;
    int digit2 = 0;
    int numberOfDigits = 0;
    int sumOdds2x = 0;
    int sumEvens = 0;

    while (creditCardNumber > 0)
    {
        digit2 = digit1;
        digit1 = creditCardNumber % 10;

        if (numberOfDigits % 2 == 0)
        {
            sumEvens += digit1;
        }
        else
        {
            int multiple = 2 * digit1;
            sumOdds2x += (multiple / 10) + (multiple % 10);
        }

        creditCardNumber /= 10;
        numberOfDigits++;
    }

    bool is_valid = (sumEvens + sumOdds2x) % 10 == 0;
    int first_two_digits = (digit1 * 10) + digit2;

    if (digit1 == 4 && numberOfDigits >= 13 && numberOfDigits <= 16 && is_valid)
    {
        printf("VISA\n");
    }
    else if ((first_two_digits == 34 || first_two_digits == 37) && numberOfDigits == 15 && is_valid)
    {
        printf("AMEX\n");
    }
    else if (first_two_digits >= 51 && first_two_digits <= 55 && numberOfDigits == 16 && is_valid)
    {
        printf("MASTERCARD\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
